### PR TITLE
Add documentation on storage options.

### PR DESCRIPTION
**Pull request by will-cromar**

### CODE REVIEW

**Comment by will-cromar** [comment]
I'm avoiding mentioning GCE PDs, since we had trouble managing those for our tests.

File: doc/storage.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  resources:
+    requests:
+      storage: 1Ki
+```
+
+`Job`s that rely on the dataset in Filestore will need to have the following line in their pod template to bind to that `PersistentVolumeClaim`:
+
+```yaml
+  ...
+  - name: filestore-datasets
```
Comment (zcain117) [inline doc/storage.md:46]: I'd add a comment on this line like "# `name` should match the `name` of your `PersistentVolume`"

(if that comment is accurate. Maybe the `name` on this line doesn't matter?)

File: doc/storage.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    persistentVolumeClaim:
+      claimName: filestore-datasets-claim
```
Comment (zcain117) [inline doc/storage.md:48]: comment here too, something like:  "# This should match the `name` of the PersistentVolumeClaim mentioned above."

File: doc/storage.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  ...
+```
```
Comment (zcain117) [inline doc/storage.md:50]: I'd add a line after this config snippet on how to refer to the dataset. In the test command that the user writes, what is the path to access the data?

File: doc/storage.md (hunk 1 above)
Comment (will-cromar) [reply doc/storage.md:46]: `name` doesn't matter here. I added a note above `claimName`.

File: doc/storage.md (hunk 2 above)
Comment (will-cromar) [reply doc/storage.md:49]: Done.

File: doc/storage.md (hunk 3 above)
Comment (will-cromar) [reply doc/storage.md:51]: Done.

**Pull request merged**